### ISSUE TITLE
iOS app freezes when trying to disclose an attribute to a server running in development mode on my laptop

### ISSUE THREAD

**Issue by tdons** [privacybydesign/irmamobile#6]
**Problem**

The IRMA iOS app freezes when I try to disclose `pbdf.nijmegen.personalData.fullname` to a local `irmago` server running in development mode (see below for versions of software used).

**Expectation**

That I can request the attribute.

**Reproduce**

1. run `./irma-master-darwin-amd64 server -v` in terminal window A
2. run `./irma-master-darwin-amd64 session --server http://localhost:8088 --disclose pbdf.nijmegen.personalData.fullname` in terminal window B
3. launch the IRMA app and turn on developer mode (by tapping 7x on the version label)
4. go back to the main screen of the IRMA app and click "QR scan" to scan the QR code produced in step 2.
5. Notice that the app freezes.  The screen remains white.  A restart of the app is required (killing it) before it goes back to normal.

**Note**

I have tried to request another attribute and that worked great. It appears as though the specific attribute I'm requesting matters.   For example: when I request `pbdf.pbdf.mobilenumber.mobilenumber` using the same steps it works fine and the attribute appears in my terminal and the app keeps functioning.  I've also tried deleting and re-adding the BRP data but that didn't help.

**Software versions used**

- `irma-master-darwin-amd64` with shasum=72618081e5c125995e168a71a546a05df0ab761f (latest master downloaded today). To make the confusion about which version I'm using complete, the shasum I mentioned doesn't refer to a git commit but is the hash of the file.
- `iOS app` version ad77578c (running in developer mode)

**Comment by davidv1992** [comment]
Thank you for the report, I have managed to recreate this locally and will look into the cause

**Comment by davidv1992** [comment]
The underlying cause of this is that the pbdf.nijmegen.personalData credential is deprecated (it has been replaced by pbdf.gemeente.personalData). A bug is causing the app to crash when processing a request for a deprecated credential without an option for when you dont have it, if your app doesn't have an instance of that credential. We have placed this on our todo list and hope to fix it reasonably soon.
